### PR TITLE
Asimov fixes

### PR DESCRIPTION
**Pull request by nihargupte-ph**
This PR adds a few new features to the asimov interface. There are two main updates 

1). Adding documentation including sample .yaml files
2). Updating the dingo.ini template file to accept most arguments 

TODO:
Add a way to specify event priors in the event_dingo.yaml file.

### CODE REVIEW

**Comment by nihargupte-ph** [comment]
It seems like currently, the way to specify the the template location of external pipelines is not 
different from Bilby and other internal pipelines. This behavior is being addressed in the following MR https://git.ligo.org/asimov/asimov/-/merge_requests/133 by @ka-lok.lo. However, as this is not merged into the latest version of asimov, I have left this line: https://github.com/dingo-gw/dingo/blob/093e5c1b4436a284d3325fcdc98fa0a12e595d40/dingo/asimov/asimov.py#L29 commented out

**Review by stephengreen — APPROVED** [review]
Looks great, thanks Nihar, especially for taking the time to write the docs!

There are just a couple comments, which should be easy to address.

File: dingo/pipe/main.py
[the remark this answers is not in the document]
Comment (nihargupte-ph) [reply dingo/pipe/main.py:1]: After reading this page: https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/proposing-changes-to-your-work-with-pull-requests/about-comparing-branches-in-pull-requests I think I get what is happening. We are seeing the three dot commit which shows the difference between the latest commit on this branch with the _latest common ancester_ commit (where this branch diverged from main). So even though the files are the same, the diff is being compared with an older file. To fix this, we can merge main into the feature branch which will update the latest common ancestor to be the current state of main.

File: dingo/pipe/data_generation.py [hunk 1]
```diff
@@ -154,8 +154,10 @@ def __init__(self, args, unknown_args, create_data=True):
             args.injection_numbers = None
             args.injection_file = None
             args.injection_dict = None
-            args.gaussian_noise = False
             args.injection_waveform_arguments = None
+            args.injection_frequency_domain_source_model = None
+            self.frequency_domain_source_model = None
+            self.gaussian_noise = False
```
[the remark this answers is not in the document]
Comment (nihargupte-ph) [reply dingo/pipe/data_generation.py]: See above comment, essentially this file is the same as it is in main now.

File: docs/source/asimov.md
[the remark this answers is not in the document]
Comment (nihargupte-ph) [reply docs/source/asimov.md:1]: True, although I think this is more a job in the asimov documentation. Maybe in a future PR we can improve this.

**Pull request merged**